src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1326,24 +1326,13 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
     /**
      * Enables time travelling for the current query (only supported by cockroach currently)
      */
-    timeTravelQuery(timestampFn?: string | false): this {
-        if (
-            timestampFn &&
-            this.connection.driver.options.type !== "cockroachdb"
-        ) {
-            throw new TypeORMError(
-                `SelectQueryBuilder.timeTravelQuery is only supported by cockroachdb.`,
-            )
-        }
-
-        if (timestampFn !== false) {
-            this.expressionMap.timeTravel = true
-
-            if (timestampFn) {
-                this.expressionMap.timeTravelQueryTimestampFn = timestampFn
+    timeTravelQuery(timeTravelFn?: string | boolean): this {
+        if (this.connection.driver.options.type === "cockroachdb") {
+            if (timeTravelFn === undefined) {
```

Comment:
My only problem with this is that it create further complexity to the logic, if you wish to enable time travelling queries based on a condition.

E.g. it's not possible to do:

```
queryBuilder.timeTravelQuery(false) // disable all TTQ
queryBuilder.timeTravelQuery(shouldUseTTQ) // disable all TTQ based on condition
```

Reply:
Ah, well, I just briefly went through the code, and it seemed like it wasn't possible to do that, but if that's possible, well then that's great!